### PR TITLE
Update switch role modules

### PR DESCRIPTION
**Pull request by eshulman2**
- updated switch module for Junos switches
- allow ipv6 on l3 interfaces

### CODE REVIEW

File: galaxy.yml [hunk 1]
```diff
@@ -81,7 +81,7 @@ dependencies:
   'git+https://github.com/openstack/ansible-collections-openstack': 'master'
   'git+https://github.com/ansible-collections/ansible.netcommon': 'main'
   'git+https://github.com/openstack/ansible-config_template': 'master'
-  'git+https://github.com/ansible-collections/junipernetworks.junos': '2.10.0'
+  'git+https://github.com/ansible-collections/junipernetworks.junos': 'main'
```
Comment (lewisdenny) [inline galaxy.yml:84]: Please also update this in the requirements:

https://github.com/openstack-k8s-operators/ci-framework/blob/00bdc9430c01230d6965015274d72af035991cb3/requirements.yml#L39

**Review by lewisdenny — CHANGES_REQUESTED** [review]

File: galaxy.yml (hunk 1 above)
Comment (eshulman2) [reply galaxy.yml:84]: done

**Review by lewisdenny — APPROVED** [review]
/lgtm

File: roles/switch_config/tasks/ios_config.yml [hunk 2]
```diff
@@ -14,11 +14,15 @@
   when: "'dhcp_relay' in switch_vars"
 
 - name: Set VLANs
-  cisco.ios.ios_vlan:
-    name: "vlan{{ item }}"
-    vlan_id: "{{ item }}"
-    state: present
-  loop: "{{ vlan_list }}"
+  vars:
+    vlan_config: |
```
Comment (pablintino) [inline roles/switch_config/tasks/ios_config.yml:18]: You can avoid the yaml conversion by simple tweaking the jinja part:

```yaml
  vars:
    vlan_config: |-
      [   
      {% for vlan in vlan_list %}
      - name: "vlan{{ vlan }}"
        vlan_id: "{{ vlan }}"
      {% endfor %}
      ]
```
With that in place you can directly pass the var like this:
```yaml
  cisco.ios.ios_vlans:
    config: "{{ vlan_config }}"
```

Or even simpler:
```yaml
  cisco.ios.ios_vlans:
    config: |-
      [   
      {% for vlan in vlan_list %}
      - name: "vlan{{ vlan }}"
        vlan_id: "{{ vlan }}"
      {% endfor %}
      ]
    state: merged

```

Comment (eshulman2) [reply roles/switch_config/tasks/ios_config.yml:18]: I tried doing semilar things but for some reason the module failed to accept them, not sure if this is a specific issue to the modules but unfortunately I was unable to find a better way to implement it with the time I had to work on it
```
fatal: [nfv_private_sw02]: FAILED! => changed=false 
  invocation:
    module_args:
      config: []
      running_config: null
      state: merged
  msg: dictionary requested, could not parse JSON or key=value
```

**Pull request merged**